### PR TITLE
feat: improve display of releases in navbar

### PR DESCRIPTION
**Pull request by tuananh**
Signed-off-by: Tuan Anh Tran <me@tuananh.org>

**1. Issue, if available:**

#826 

**2. Description of changes:**

show the latest version in navbar + style pre-release in gray color

image

**3. Does this change impact docs?**
- [ ] Yes, PR includes docs updates
- [ ] Yes, issue opened: *link to issue*
- [ ] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by tuananh** [comment]
@geoffcline can you take a look?

File: website/config.toml [hunk 1]
```diff
@@ -143,3 +140,6 @@ pre = "<i class='fab fa-github'></i>"
   version = "v0.4.3"
   url = "https://karpenter.sh/v0.4.3-docs/"
 
+[[params.versions]]
+  version = "(Pre-release)"
+  url = "https://karpenter.sh/docs/"
```
Comment (geoffcline) [inline website/config.toml:145]: should we always have a "pre-release-docs" version going? we don't currently.

Comment (geoffcline) [reply website/config.toml:145]: I'm leaning towards enabling pre-release docs only as needed.

**Review by geoffcline — CHANGES_REQUESTED** [review]

File: website/config.toml [hunk 2]
```diff
@@ -76,7 +76,8 @@ images = ["banner.png"]
 # Version configuration
 # Menu title if your navbar has a versions selector to access old versions of your site.
 # This menu appears only if you have at least one [params.versions] set.
-version_menu = "Releases"
+version_menu = "v0.5.0"
+
 # Flag used in the "version-banner" partial to decide whether to display a
```
[the remark this answers is not in the document]
Comment (tuananh) [reply website/config.toml:81]: lemme check docsy doc how do they handle it :)

File: website/config.toml (hunk 1 above)
Comment (tuananh) [reply website/config.toml:145]: i thought the docs in main branch is the pre-release :)) my bad if it's not what it is

File: website/assets/scss/_variables_project.scss [hunk 3]
```diff
@@ -66,3 +66,6 @@ nav.navbar-dark {
   }
 }
 
+.dropdown-menu .dropdown-item:nth-last-child(1) {
+  color: gray;
+}
```
[the remark this answers is not in the document]
Comment (tuananh) [reply website/assets/scss/_variables_project.scss:71]: yep. lemme find a way to fix it

[the remark this answers is not in the document]
Comment (tuananh) [reply website/assets/scss/_variables_project.scss:71]: looking at docsy theme, there's no id or specific class for this releases dropdown, we may have to either fork it or patch it before building (which is rather ugly hack IMO)

**Comment by tuananh** [comment]
@geoffcline 

1. i did a hack by patching docsy theme. Lemme see if i can upstream this change so we don't have to keep the patch here. basically, what it does is trying to get the path dir and see if it matches with any of the docs folder and display the {{ .version }} there.
2. i remove pre-release as well as the gray out css bit. since we dont have pre-release, so its' not needed right?

when browsing in /docs/....
image

when browsing in older docs
image

**Comment by tuananh** [comment]
i think i need to update netlify.toml too to apply the patch for preview.

**Comment by akestner** [comment]
Will this ensure that the docs version dropdown is visible on mobile devices as well?

**Comment by tuananh** [comment]
> Will this ensure that the docs version dropdown is visible on mobile devices as well?

no. the navbar is kind of jammed right now. in order to show version, we may have to hide other links. what do you think we should hide under hamburger menu?

**Comment by tuananh** [comment]
> Will this ensure that the docs version dropdown is visible on mobile devices as well?

it should show on mobile as well now

image

**Comment by rothgar** [comment]
This lgtm. Is the pre-release version added to config.toml manually when we have one to deploy?

I don't see any problem carrying the patch in layouts/partials because that's where you're supposed to modify/extend docsy anyway. It would be nice to be an option upstream but I don't think it should block this PR if it's not there.

**Comment by chrisnegus** [comment]
@geoffcline have all your issues been addressed. If so, /lgtm from me.

**Comment by geoffcline** [comment]
Wonderful, thank you!

**Review by geoffcline — APPROVED** [review]

**Pull request merged**